### DISCUSSION TITLE
Makefile not properly escaped for windows ?

### DISCUSSION THREAD

**Discussion by NicolasEsseiva** [cocotb/cocotb#3946] [Q&A] [answered]
I tried to use cocotb from windows 11, using miniconda.
(conda 24.4.0, cocotb 1.8.1, modelsim)

First issue : 
```
(py3_32) C:\workspace\cocotb_test>make SIM=modelsim
bash: C:Program: command not found
bash: - : invalid option
Usage:  bash [GNU long option] [option] ...
        bash [GNU long option] [option] script-file ...
GNU long options:
        --debug
        --debugger
        --dump-po-strings
        --dump-strings
        --help
        --init-file
        --login
        --noediting
        --noprofile
        --norc
        --posix
        --protected
        --rcfile
        --restricted
        --verbose
        --version
        --wordexp
Shell options:
        -ilrsD or -c command or -O shopt_option         (invocation only)
        -abefhkmnptuvxBCHP or -o option
C:\...\cocotb\share\makefiles/Makefile.inc:79: *** Unknown Python architecture: .  Stop.
```

I fixed this by specifying in the environment variables PYTHON_BIN : `C:\Users\xxx\miniconda3\envs\py3_32\python.exe`

Second issue : 
```
(py3_32) C:\workspace\cocotb_test>make SIM=modelsim
bash: C:Program: command not found
bash: C:Usersnicolminiconda3envspy3_32python.exe: command not found
C:\Users\nicol\AppData\Roaming\Python\Python312\site-packages\cocotb\share\makefiles/Makefile.inc:79: *** Unknown Python architecture: .  Stop.
```

The path is not properly escaped so I hacked this by adding the following commands in the makefile.inc :
```
win_to_unix = $(shell echo '$1' | sed -e 's/\\/\\\\/g')
PYTHON_BIN := $(call win_to_unix,$(PYTHON_BIN))
```

And finally, I get the 'Quickstart Guide' (https://docs.cocotb.org/en/stable/quickstart.html) to work

Is that normal behaviour ? Am I doing something wrong ?

**Comment by ktbarrett** [comment, marked as answer]
You aren't doing anything wrong. Something changed in the msys2 packages a couple years ago and I havent been able to get conda/msys2 to work anymore, but it works in CI? Idk. For 2.0 we are going to suggest Windows users switch to [the Python runner](https://docs.cocotb.org/en/stable/runner.html). Using the Python runner in the conda/msys2 environment *should* work right now.

**Comment by NicolasEsseiva** [comment]
I'm not really familiar with CI/CD and even less if it would require questa.

I tried the Python Runner as you suggested, removed the PYTHON_BIN environment variable, so using default (C:\Program Files\Python312\python.exe) without conda/msys2

For now it works perfectly fine, no issue.
Thanks a lot for that, I will then close this issue

**Reply by ktbarrett** [reply]
You can also try using WSL, we found this is the option with the least issues on Windows. Just make sure that you install a Linux version of Questa if you do.
